### PR TITLE
Feature: dropdown insted of tabs for orders and charts headers

### PR DESCRIPTION
**Pull request by Kohelbekker**

### CODE REVIEW

File: src/components/TabPanel/index.tsx [hunk 1]
```diff
@@ -79,18 +85,55 @@ export class TabPanel extends React.Component<TabPanelProps> {
         const navCx = 'cr-tab-panel__navigation-container-navigation';
 
         return (
-            <div className={className}>
-                <div className="cr-tab-panel__navigation-container draggable-container">
-                    <div className={navCx} role="tablist">
-                        {panels.map(this.renderTabPanel)}
+            <React.Fragment>
+                {!dropdown ? (
```
Comment (oyershov) [inline src/components/TabPanel/index.tsx:89]: 1. Need to check for truth (!dropdown -> dropdown)
2. Next code is unnecessary duplicated. Need to render conditionally only "..." part. Can be defined additional method for this.
```
<div className={className}>
    <div className="cr-tab-panel__navigation-container draggable-container">
          ...
          {optionalHead && <div className="cr-tab-panel__optinal-head">{optionalHead}</div>}
     </div>
     {contents}
</div>

File: src/components/TabPanel/index.tsx [hunk 2]
```diff
@@ -79,18 +85,55 @@ export class TabPanel extends React.Component<TabPanelProps> {
...
+                )}
+            </React.Fragment>
         );
     }
 
+    private dropdownLabels = () => {
+        const { panels, currentTabIndex } = this.props;
+
+        const tabNames = panels.map(panel => panel.label).filter(label => label !== panels[currentTabIndex].label);
+        tabNames.unshift(panels[currentTabIndex].label);
```
Comment (oyershov) [inline src/components/TabPanel/index.tsx:118]: always put checks for existing object. If "panels" is [], will raise an error
Снимок экрана 2020-08-31 в 10 35 38

File: src/components/TabPanel/index.tsx [hunk 3]
```diff
@@ -79,18 +85,55 @@ export class TabPanel extends React.Component<TabPanelProps> {
...
                     </div>
-                    {optionalHead && <div className="cr-tab-panel__optinal-head">{optionalHead}</div>}
-                </div>
-                {contents}
-            </div>
+                ) : (
+                    <div className={className}>
+                        <div className="cr-tab-panel__navigation-container draggable-container">
+                            <div className="cr-tab-panel__dropdown">
+                                <DropdownComponent list={this.dropdownLabels()} className="cr-tab-panel__dropdown" onSelect={this.handleOrderTypeChange} placeholder=""/>
```
Comment (oyershov) [inline src/components/TabPanel/index.tsx:103]: List must use transltions

File: src/components/TabPanel/index.tsx [hunk 4]
```diff
@@ -79,18 +85,55 @@ export class TabPanel extends React.Component<TabPanelProps> {
...
+
+        const activeIndex = panels.findIndex(tab => tab.label === currentLabels[index]);
+
+        if (this.props.onCurrentTabChange) {
+            this.props.onCurrentTabChange(activeIndex);
+        }
+        if (this.props.onTabChange) {
+            this.props.onTabChange(activeIndex, panels[activeIndex].label);
+        }
+    };
```
Comment (oyershov) [inline src/components/TabPanel/index.tsx:135]: reuse here ```createOnTabChangeHandler``` method

File: src/containers/RecentTrades/RecentTrades.pcss [hunk 5]
```diff
@@ -64,6 +64,70 @@
...
     }
 
+    &__dropdown {
+      .cr-dropdown {
+        height: 34px;
+
+        .dropdown-toggle,
+        .dropdown-toggle:focus,
+        .dropdown-toggle:active {
+          color: var(--primary-cta-color) !important;
```
Comment (oyershov) [inline src/containers/RecentTrades/RecentTrades.pcss:74]: Avoid to use use ```!important``` (use correct nesting to apply styles). We need to override stiles in future.
All places in this file.

File: src/containers/RecentTrades/index.tsx [hunk 6]
```diff
@@ -70,13 +72,16 @@ class RecentTradesComponent extends React.Component<RecentTradesProps, State> {
...
+        const { isMobileDevice } = this.props;
+
         return this.props.userLoggedIn ?
         (
             <TabPanel
                 panels={this.renderTabs()}
                 onTabChange={this.handleMakeRequest}
                 optionalHead={this.props.intl.formatMessage({ id: 'page.body.trade.header.recentTrades' })}
                 currentTabIndex={this.state.index}
+                dropdown={isMobileDevice}
```
Comment (oyershov) [inline src/containers/RecentTrades/index.tsx:84]: Naming is wierd. "isMobileDevice" isn't "dropdown"😅
just pass "isMobileDevice" as it is.

File: src/mobile/styles/components/Orders.pcss [hunk 7]
```diff
@@ -56,6 +56,70 @@
       }
     }
 
+    &__dropdown {
```
Comment (oyershov) [inline src/mobile/styles/components/Orders.pcss:59]: oh...
Remove duplicating styles. Just change ```Dropdown.pcss``` for mobile, with additional class name if it's needed, but don't duplicate such things.

File: src/components/TabPanel/index.tsx [hunk 8]
```diff
@@ -79,18 +85,55 @@ export class TabPanel extends React.Component<TabPanelProps> {
...
+                            {optionalHead && <div className="cr-tab-panel__optinal-head">{optionalHead}</div>}
+                        </div>
+                        {contents}
+                    </div>
+                )}
+            </React.Fragment>
         );
     }
 
+    private dropdownLabels = () => {
```
Comment (oyershov) [inline src/components/TabPanel/index.tsx:114]: try to use ```React.memo()``` to avoid unexpected re-rendering

**Review by oyershov — CHANGES_REQUESTED** [review]

File: src/components/TabPanel/index.tsx (hunk 3 above)
Comment (Kohelbekker) [reply src/components/TabPanel/index.tsx:103]: But list always use panels.label values. And panels have translations here - https://github.com/openware/baseapp/blob/feature/trading-page-dropdown/src/mobile/components/Orders/index.tsx#L70

**Pull request merged**